lms/djangoapps/course_structure_api/v0/tests.py

```diff
@@ -0,0 +1,329 @@
+            end=datetime(2015, 1, 16),
+            org="MTD"
+        )
+
+    def build_absolute_url(self, path=None):
+        """ Build absolute URL pointing to test server.
+        :param path: Path to append to the URL
+        """
+        url = TEST_SERVER_HOST
+
+        if path:
+            url += path
+
+        return url
+
+    def assertValidResponseCourse(self, data, course):
+        """ Determines if the given response data (dict) matches the specified course. """
+
+        course_key = course.id
+        self.assertEqual(data['id'], unicode(course_key))
+        self.assertEqual(data['name'], course.display_name)
+        self.assertEqual(data['course'], course_key.course)
+        self.assertEqual(data['org'], course_key.org)
+        self.assertEqual(data['run'], course_key.run)
+
+        uri = self.build_absolute_url(
+            reverse('course_structure_api:v0:detail', kwargs={'course_id': unicode(course_key)}))
+        self.assertEqual(data['uri'], uri)
+
+    def http_get(self, uri, **headers):
+        """Submit an HTTP GET request"""
+
+        default_headers = {
+            'HTTP_AUTHORIZATION': 'Bearer ' + self.access_token
+        }
+        default_headers.update(headers)
+
+        response = self.client.get(uri, content_type='application/json', follow=True, **default_headers)
+        return response
+
+    def test_not_authenticated(self):
+        """
+        Verify that access is denied to non-authenticated users.
+        """
+        raise NotImplementedError
+
+    def test_not_authorized(self):
+        """
+        Verify that access is denied to non-authorized users.
+        """
+        raise NotImplementedError
+
+
+class CourseDetailMixin(object):
+    """
+    Mixin for views utilizing only the course_id kwarg.
+    """
+
+    def test_get_invalid_course(self):
+        """
+        The view should return a 404 if the course ID is invalid.
+        """
+        response = self.http_get(reverse(self.view, kwargs={'course_id': self.invalid_course_id}))
+        self.assertEqual(response.status_code, 404)
+
+    def test_get(self):
+        """
+        The view should return a 200 if the course ID is invalid.
+        """
+        response = self.http_get(reverse(self.view, kwargs={'course_id': self.course_id}))
+        self.assertEqual(response.status_code, 200)
+
+        # Return the response so child classes do not have to repeat the request.
+        return response
+
+    def test_not_authenticated(self):
+        # If debug mode is enabled, the view should always return data.
+        with override_settings(DEBUG=True):
+            response = self.http_get(reverse(self.view, kwargs={'course_id': self.course_id}), HTTP_AUTHORIZATION=None)
+            self.assertEqual(response.status_code, 200)
+
+        response = self.http_get(reverse(self.view, kwargs={'course_id': self.course_id}), HTTP_AUTHORIZATION=None)
+        self.assertEqual(response.status_code, 403)
```

Comment:
Fixed. I had to workaround the fact that DRF does not actually enforce authentication. Instead it simply passes along an `AnonymousUser`. I overrode `perform_authentication` to raise `AuthenticationFailed` when an `AnonymousUser` is encountered.